messaging/src/main/java/org/axonframework/usage/UpdateCheckTask.java

```diff
@@ -0,0 +1,216 @@
+ * available upgrades and vulnerabilities in the Axon libraries used. These vulnerabilities and upgrades are logged.
+ * <p>
+ * The task is scheduled to run periodically, with an initial delay of 1 second. The interval between runs is determined
+ * by the response from the telemetry endpoint.
+ * <p>
+ * The task will retry reporting in case of an error, with an exponential backoff strategy. The backoff factor is
+ * increased with each failed attempt, up to a maximum of 60 seconds. The first request is sent as a POST request,
+ * subsequent requests are sent as PUT requests.
+ * <p>
+ * The task will not run if the user has opted out of anonymous usage reporting. See {@link UpdateChecker} for
+ * more details on how the opt-out is determined.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+@Internal
+public class UpdateCheckTask implements Runnable {
+    private final HttpClient client;
+    private final UsagePropertyProvider userProperties;
+    private int errorRetryBackoffFactor = 1;
+    private boolean firstRequest = true;
+
+    /**
+     * Starts the anonymous usage reporting task. If the user has opted out of anonymous usage reporting, the task will
+     * not be started.
+     */
+    public static void start() {
+        try {
+            UsagePropertyProvider userProperties = UsagePropertyProvider.create();
+            if (userProperties.getDisabled()) {
+                logger.info("Anonymous usage reporting is opted out by the user. Skipping task initialization.");
+                return;
+            }
+            DelayedTask.of(new UpdateCheckTask(userProperties), 1000);
+        } catch (Exception e) {
+            logger.warn("Failed to start Anonymous Usage Collector task.", e);
+        }
+    }
+
+    private UpdateCheckTask(UsagePropertyProvider userProperties) {
+        this.userProperties = userProperties;
+        this.client = HttpClient.newBuilder()
+                                .followRedirects(HttpClient.Redirect.ALWAYS)
+                                .build();
+    }
+
+    @Override
+    public void run() {
+        try {
+            UsageRequest requestBody = buildRequest();
+            String serializedRequestBody = requestBody.serialize();
+            logger.debug("Reporting anonymous usage data:\n{}", serializedRequestBody);
+
+            HttpRequest.Builder baseRequestBuilder = HttpRequest.newBuilder()
+                                                                .uri(URI.create(userProperties.getUrl()))
+                                                                .timeout(Duration.ofSeconds(5))
+                                                                .headers("Content-Type", "text/plain;charset=UTF-8");
+
+            HttpRequest request;
+            if (firstRequest) {
+                request = baseRequestBuilder
+                        .POST(HttpRequest.BodyPublishers.ofString(serializedRequestBody))
```

Comment:
We might want to consider using GET instead. Long story, and I expect to get confirmation tomorrow afternoon. The URL parameters of the GET request can be 4k without problems. Officially, there is no limit, but security recommendations talk about maximizing to avoid DDOS attacks.